### PR TITLE
Refined settlement-related details

### PR DESCRIPTION
**Pull request by vandaillyes-modusbox**

### CODE REVIEW

**Review by elnyry-sam-k — DISMISSED** [review]

File: admin-api/central_ledger_api_v1.0.json [hunk 1]
```diff
@@ -1960,7 +1985,7 @@
                 "type": "string",
                 "minLength": 1,
                 "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$",
-                "description": "Quote Identifier. The API data type UUID (Universally Unique Identifier) is a JSON String in canonical format, conforming to RFC 4122, that is restricted by a regular expression for interoperability reasons. An UUID is always 36 characters long, 32 hexadecimal symbols and 4 dashes (‘-‘). A minLength of 1 has been specified due to the regex not being validated on empty strings, this should be removed once the enjoi module has been fixed."
+                "description": "Quote Identifier. The API data type UUID (Universally Unique Identifier) is a JSON String in canonical format, conforming to RFC 4122, that is restricted by a regular expression for interoperability reasons. An UUID is always 36 characters long, 32 hexadecimal symbols and 4 dashes (‘-‘). A minLength of 1 has been specified due to the regex not being validated on empty strings, this should be removed once the joi module has been fixed."
```
Comment (mdebarros) [inline admin-api/central_ledger_api_v1.0.json:1988]: Please remove `, this should be removed once the joi module has been fixed.` <-- I don't think its applicable here.

File: admin-api/central_ledger_api_v1.0.json [hunk 2]
```diff
@@ -2086,7 +2114,7 @@
                 "type": "string",
                 "minLength": 1,
                 "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$",
-                "description": "Transaction Identifier. The API data type UUID (Universally Unique Identifier) is a JSON String in canonical format, conforming to RFC 4122, that is restricted by a regular expression for interoperability reasons. An UUID is always 36 characters long, 32 hexadecimal symbols and 4 dashes (‘-‘). A minLength of 1 has been specified due to the regex not being validated on empty strings, this should be removed once the enjoi module has been fixed."
+                "description": "Transaction Identifier. The API data type UUID (Universally Unique Identifier) is a JSON String in canonical format, conforming to RFC 4122, that is restricted by a regular expression for interoperability reasons. An UUID is always 36 characters long, 32 hexadecimal symbols and 4 dashes (‘-‘). A minLength of 1 has been specified due to the regex not being validated on empty strings, this should be removed once the joi module has been fixed."
```
Comment (mdebarros) [inline admin-api/central_ledger_api_v1.0.json:2117]: Please remove `, this should be removed once the joi module has been fixed.` <-- I don't think its applicable here.

**Review by mdebarros — CHANGES_REQUESTED** [review]
Thanks @vandaillyes-modusbox.

I have 2 comments for your consideration.

I am also seeing a lot of repetition in the swagger. I think these should be moved to external types and references appropriately. Not sure if that should be done as part of this PR, or as a future enhancement. @sam what do you think?

That will help with the maintenance since we will only have a single description/definition for each of these types.

**Comment by elnyry-sam-k** [comment]
>I am also seeing a lot of repetition in the swagger. I think these should be moved to external types and references appropriately. Not sure if that should be done as part of this PR, or as a future enhancement. @sam what do you think?

Thanks Miguel, @mdebarros, as long as these do not affect implementation side of things, we can go ahead with the change.. For those changes that need implementation changes, lets do in the next iteration...

File: admin-api/central_ledger_api_v1.0.json (hunk 1 above)
Comment (vandaillyes-modusbox) [reply admin-api/central_ledger_api_v1.0.json]: OK, removed it.

File: admin-api/central_ledger_api_v1.0.json (hunk 2 above)
Comment (vandaillyes-modusbox) [reply admin-api/central_ledger_api_v1.0.json]: OK, removed it.

File: admin-api/central_ledger_api_v1.0.json [hunk 3]
```diff
@@ -2050,71 +1804,105 @@
...
                             "INTERCHANGE_FEE",
-                            "POSITION"
+                            "POSITION",
+                            "SETTLEMENT"
                         ]
                     },
+                    "settlementAccountType": {
+                        "type": "string",
+                        "description": "A special type of ledger account into which settlements should be settled: SETTLEMENT or INTERCHANGE_FEE_SETTLEMENT.",
+                        "enum": [
```
Comment (mdebarros) [inline admin-api/central_ledger_api_v1.0.json:1833]: 1. I think we should remove the enum restriction on this as it's possible to add additional settlementAccountTypes via the admin API, and those accounts can be handled by a custom Settlement Rule.

Rather change `enum` to `examples` as per the following example:

```
  "examples": [
      "INTERCHANGE_FEE",
      "SETTLEMENT"
  ]
```

2. Remove`POSITION` from the description & example as it is not a valid settlementAccountType. For settling the position, the settlementAccountType `SETTLEMENT` will be used.

**Review by mdebarros — CHANGES_REQUESTED** [review]
Looks good, great work remove the duplicate definitions!

Minor suggestions comment for your attention.

**Review by mdebarros — APPROVED** [review]
+1

**Review by elnyry-sam-k — APPROVED** [review]

**Pull request merged**